astropy/io/fits/tests/test_core.py

```diff
@@ -1062,20 +1056,7 @@ def tell(self):
                         if hdu1.data is not None and hdu2.data is not None:
                             assert np.all(hdu1.data == hdu2.data)
 
-    @pytest.mark.skipif('not HAVE_STRINGIO')
-    def test_write_stringio(self):
-        """
-        Regression test for https://github.com/astropy/astropy/issues/2463
-
-        Only test against `StringIO.StringIO` on Python versions that have it.
-        Note: `io.StringIO` is not supported for this purpose as it does not
-        accept a bytes stream.
-        """
-
-        self._test_write_string_bytes_io(StringIO.StringIO())
-
-    @pytest.mark.skipif('not HAVE_STRINGIO')
-    def test_write_stringio_discontiguous(self):
+    def test_write_bytesio_discontiguous(self):
```

Comment:
not sure if it would be better to remove that test instead of using `BytesIO` here.

Reply:
Having a test with `BytesIO` is useful but maybe it's already tested elsewhere ?

Reply:
I haven't seen one that explicitly tested writing an discontiguous array. But there are several writing to `BytesIO`.